Interfaces/IFluxImage.cs

```diff
@@ -0,0 +1,94 @@
+//
+//     Defines the interface to be implemented by flux image plugins.
+//
+// --[ License ] --------------------------------------------------------------
+//
+//     Permission is hereby granted, free of charge, to any person obtaining a
+//     copy of this software and associated documentation files (the
+//     "Software"), to deal in the Software without restriction, including
+//     without limitation the rights to use, copy, modify, merge, publish,
+//     distribute, sublicense, and/or sell copies of the Software, and to
+//     permit persons to whom the Software is furnished to do so, subject to
+//     the following conditions:
+//
+//     The above copyright notice and this permission notice shall be included
+//     in all copies or substantial portions of the Software.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+//     OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//     MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//     IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+//     CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+//     TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+//     SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// ----------------------------------------------------------------------------
+// Copyright © 2011-2023 Rebecca Wallander
+// ****************************************************************************/
+
+using Aaru.CommonTypes.Enums;
+
+namespace Aaru.CommonTypes.Interfaces;
+
+/// <inheritdoc />
+/// <summary>Abstract class to implement flux reading plugins.</summary>
+public interface IFluxImage : IBaseImage
+{
+    /// <summary>
+    ///     An image may have more than one capture for a specific head/track/sub-track combination. This returns
+    ///     the amount of captures in the image for the specified head/track/sub-track combination.
+    /// </summary>
+    /// <returns>The number of captures</returns>
+    /// <param name="head">Physical head (0-based)</param>
+    /// <param name="track">Physical track (position of the heads over the floppy media, 0-based)</param>
+    /// <param name="subTrack">Physical sub-step of track (e.g. half-track)</param>
+    uint CapturesLength(uint head, ushort track, byte subTrack);
+
+    /// <summary>Reads the resolution (sample rate) of a flux capture in picoseconds</summary>
+    /// <returns>The resolution of a capture in picoseconds</returns>
+    /// <param name="head">Physical head (0-based)</param>
+    /// <param name="track">Physical track (position of the heads over the floppy media, 0-based)</param>
+    /// <param name="subTrack">Physical sub-step of track (e.g. half-track)</param>
+    /// <param name="captureIndex">Which capture to read. See also <see cref="CapturesLength" /></param>
+    ulong ReadFluxResolution(uint head, ushort track, byte subTrack, uint captureIndex);
```

Comment:
What if the specified h/t/s/idx parameters do not exist? It should return an ErrorNumber indicating so.